compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -71,6 +71,14 @@ object ProtoTypes {
                    |constraint was: ${ctx.typerState.constraint}
                    |constraint now: ${newctx.typerState.constraint}""")
             if result && (ctx.typerState.constraint ne newctx.typerState.constraint) then
+              // Remove all type lambdas and tvars introduced by testCompat
+              for tvar <- newctx.typerState.ownedVars do
+                val tl = tvar.origin.binder
+                newctx.typerState.ownedVars -= tvar
+                if newctx.typerState.constraint.contains(tl) then
+                  newctx.typerState.constraint = newctx.typerState.constraint.remove(tl)(using newctx)
```

Comment:
I think this is correct but a bit low-level and instead we could write:
```suggestion
              // Remove all tvars introduced by testCompat
              inContext(newctx):
                for tvar <- newctx.typerState.ownedVars do
                  if !tvar.isInstantiated then
                    tvar.instantiate(fromBelow = true)
```
and let `commit()` remove the type lambda.

Reply:
Sure